### PR TITLE
feat(shorebird_cli): `shorebird apps list`

### PR DESCRIPTION
**Pull request by felangel**
## Description

- feat(shorebird_cli): add shorebird apps list

```sh
$ shorebird apps list
shorebird-counter: v1.0.0 (patch v2)
shorebird-example: (empty)
```

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/commands/apps/list_apps_command.dart [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+class ListAppsCommand extends ShorebirdCommand {
+  /// {@macro list_apps_command}
+  ListAppsCommand({
+    required super.logger,
+    super.buildCodePushClient,
+    super.auth,
+  });
+
+  @override
+  String get description => 'List all Shorebird apps.';
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/commands/apps/list_apps_command.dart:22]: I wonder if they're "shorebird apps" or "apps using Shorebird"?

File: packages/shorebird_cli/lib/src/commands/apps/list_apps_command.dart [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  String get name => 'list';
+
+  @override
+  List<String> get aliases => ['ls'];
+
+  @override
+  Future<int>? run() async {
+    final session = auth.currentSession;
+    if (session == null) {
+      logger.err('You must be logged in to publish.');
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/commands/apps/list_apps_command.dart:34]: "publish"

Comment (eseidel) [reply packages/shorebird_cli/lib/src/commands/apps/list_apps_command.dart:34]: This also feels like "middleware" at some level.  Some sort of bool on ShorebirdCommand constructor?

Comment (felangel) [reply packages/shorebird_cli/lib/src/commands/apps/list_apps_command.dart:34]: Good catch will fix now

**Pull request merged**